pkg/packer/manifests/gcp/packer.json.tmpl

```diff
@@ -0,0 +1,203 @@
+{
+  "variables": {
+    "account_file": "{{ env `GOOGLE_APPLICATION_CREDENTIALS` }}",
+    "ansible_extra_vars": "{{ user `ansible_extra_vars` }}",
+    "build_name": "",
+    "build_name_extra": "",
+    "build_timestamp": "{{ timestamp }}",
+    "gpu": "false",
+    "gpu_nvidia_version": "",
+    "gpu_types": "",
+    "konvoy_image_builder_version": "0.0.1",
+    "kubernetes_full_version": "",
+    "manifest_output": "manifest.json",
+    "network": "",
+    "project_id": "",
+    "zone": ""
+  },
+  "builders": [
+    {
+       "image_name": "{{ (user `distribution`) | lower }}-{{ user `distribution_version` }}{{ user `build_name_extra` }}",
```

Comment:
The `image_name` should be similar in format as we use in AWS where `kuberentes_version` is used.
ex. https://github.com/mesosphere/konvoy-image-builder/blob/main/pkg/packer/manifests/aws/packer.json.tmpl#L63

I think you can use this value for the `"name"` config. 
```
"name": "{{ (user `distribution`) | lower }}-{{ user `distribution_version` }}{{ user `build_name_extra` }}",
```

Reply:
So your suggestion is what we are using for image name. Im not sure I understand, azure is doing the same for the image name and I am following that convention thought I guess that is open to debate. Do you want to also include a timestamp in the name?

Reply:
There are configs:
- `name`: Name of the packer build that shows up on the packer output.
- `image_name` (gcp) | `ami_name`(aws) | `managed_image_name` (azure) | `vm_name` (vsphere): The name of the resulted image that would be stored on the cloud provider which will be used to create cluster.

we should be wanting to  include `kubernetes_version` in the image name for all providers. This will help extend automated build chains to `lookup` / `search` images based on the kubernetes version, base os and timestamp. 
The current automated build chain (k8s upstream -> rpms -> KIB -> Konvoy -> dkp) is built for AWS but this will help it to extend other providers too. The build chains are based on kubernetes version so I think the KIB image name should include k8s version to reflect that. 

I think our Azure packer config should be setup similar way too. by just reading through packer documentation, it seems like Azure should be setting proper `managed_image_name` for https://www.packer.io/plugins/builders/azure/arm#managed_image_name-1 and use similar format as AWS AMI: `"konvoy-ami-{{user `build_name`}}-{{user `kubernetes_full_version` | clean_resource_name}}-{{user `build_timestamp`}}"`

Reply:
I understand we may want to do this. However, nothing in the documentation says it is required, and even in our code we do not require this nor do we ask that the user use the mananged name in a certain format. I think you bring up some interesting points, and maybe we should reconsider these pieces as we move forward. I would be ok to change the name now, but I would rather have a conversation around this outside of this PR and then see how we proceed. If we do decide that we want a specific name, then we probably should have a default value for say gallery_image_name, which I cannot find one in the current code.